### PR TITLE
allow user to vote on article

### PR DESCRIPTION
**Pull request by DeanDevine**

### CODE REVIEW

File: src/components/Article.jsx [hunk 1]
```diff
@@ -15,6 +17,30 @@ function Article() {
...
+  const handleClickVoteDown = () => {
+    patchArticle(article_id, -1)
+      .then((updatedArticle) => {
+        setArticleVotes((currentArticleVotes) => {
+          return currentArticleVotes - 1;
+        });
+      })
+      .catch((err) => {
+        setApiError(err.response);
+      });
```
Comment (HeatherEver) [inline src/components/Article.jsx:32]: we could make this a little more DRY - as these two functions are doing the same thing it's just whether it's plus or minus 1 - think how you could just have one function!

**Review by HeatherEver — CHANGES_REQUESTED** [review]
Again good work but some things to think about: 
- the two functions i mentioned that are quite repetitive
- with slow internet it takes a while for the vote to change so optimistic rendering is needed here and think how to handle if the request failed! (this one will need changing before merging and a new pr submitting)
- when offline and the request does fail it would be useful to have some feedback for the user

File: src/components/Article.jsx [hunk 2]
```diff
@@ -17,27 +17,17 @@ function Article() {
...
-        });
-      })
-      .catch((err) => {
-        setApiError(err.response);
-      });
-  };
+  const handleClick = (event) => {
+    setArticleVotes((currentArticleVotes) => {
+      return currentArticleVotes + event;
+    });
```
Comment (MKD01) [inline src/components/Article.jsx]: Is event an appropriate name here? What is event? What kind of data does it hold?

File: src/components/api.js [hunk 3]
```diff
@@ -22,9 +22,9 @@ export const getComments = (article_id) => {
   });
 };
 
-export const patchArticle = (article_id, vote) => {
+export const patchArticle = (article_id, event) => {
   const patchRequestBody = {
-    inc_votes: vote,
+    inc_votes: event,
```
Comment (MKD01) [inline src/components/api.js]: Was there a reason you changed vote to event here?

File: src/components/Article.jsx [hunk 4]
```diff
@@ -17,27 +17,17 @@ function Article() {
...
-          return currentArticleVotes - 1;
-        });
+    patchArticle(article_id, event)
+      .then(() => {
+        setApiError(null);
       })
       .catch((err) => {
-        setApiError(err.response);
+        setApiError(err);
       });
```
Comment (MKD01) [inline src/components/Article.jsx]: If a request fails your current implementation breaks. I would suggest taking a look at the err object you get back to see what this actually looks like.

**Review by MKD01 — APPROVED** [review]
Over all great work, but there are a few things to take a look at. Once you have made all the changes im happy for you to merge this in 👍

**Pull request merged**